packages/cli/src/util/client.ts

```diff
@@ -146,10 +146,15 @@ export default class Client extends EventEmitter implements Stdio {
         const error = await responseError(res);
 
         if (isSAMLError(error)) {
-          // A SAML error means the token is expired, or is not
-          // designated for the requested team, so the user needs
-          // to re-authenticate
-          await this.reauthenticate(error);
+          try {
+            // A SAML error means the token is expired, or is not
+            // designated for the requested team, so the user needs
+            // to re-authenticate
+            await this.reauthenticate(error);
+          } catch (reauthError) {
+            // there's no sense in retrying
+            return bail(reauthError as Error);
```

Comment:
To avoid the `as` cast, import this function from `is-error.ts`.

```suggestion
            return bail(normalizeError(reauthError));
```